src/setuptools_scm/_config.py

```diff
@@ -105,6 +105,9 @@ class Configuration:
 
     parent: _t.PathT | None = None
 
+    def __post_init__(self) -> None:
+        self.tag_regex = _check_tag_regex(self.tag_regex)
```

Comment:
taking this here without leaving it in place creates potental trouble for valies valid in config but not in the config objects